code/scripting/global_hooks.cpp

```diff
@@ -33,51 +118,222 @@ const std::shared_ptr<OverridableHook> OnShipCollision = OverridableHook::Factor
+		{"Object",
+			"weapon",
+			"The weapon object with which the \"other\" object collided with. Provided for consistency with other "
+			"collision hooks."},
+		{"Weapon", "weapon", "Same as \"Object\""},
+		{"Hitpos", "vector", "The world position where the colission was detected"},
+		{"Debris", "object", "The debris object with which the ship collided (only set for debris collisions)"},
+		{"Asteroid", "object", "The asteroid object with which the ship collided (only set for asteroid collisions)"},
+		{"Ship", "ship", "The ship object with which the weapon collided (only set for ship collisions)."},
+		{"WeaponB", "weapon", "For weapon on weapon collisions, the \"other\" weapon."},
+		{"Beam", "weapon", "The beam object with which the ship collided (only set for beam collisions)"} });
+
+const std::shared_ptr<OverridableHook<CollisionConditions>> OnDebrisCollision = OverridableHook<CollisionConditions>::Factory("On Debris Collision",
+	"Invoked when a ship collides with another object.",
+	{ {"Self", "object", "The \"other\" object that collided with the ship."},
+		{"Object",
+			"debris",
+			"The debris object with which the \"other\" object collided with. Provided for consistency with other "
+			"collision hooks."},
+		{"Debris", "debris", "Same as \"Object\""},
+		{"Hitpos", "vector", "The world position where the colission was detected"},
+		{"Asteroid", "object", "The asteroid object with which the ship collided (only set for asteroid collisions)"},
+		{"Ship", "ship", "The ship object with which the ship collided (only set for ship collisions)."},
+		{"Weapon", "weapon", "The weapon object with which the ship collided (only set for weapon collisions)"},
+		{"Beam", "weapon", "The beam object with which the ship collided (only set for beam collisions)"} });
+
+const std::shared_ptr<OverridableHook<CollisionConditions>> OnAsteroidCollision = OverridableHook<CollisionConditions>::Factory("On Asteroid Collision",
+	"Invoked when a ship collides with another object.",
+	{ {"Self", "object", "The \"other\" object that collided with the ship."},
+		{"Object",
+			"object",
+			"The asteroid object with which the \"other\" object collided with. Provided for consistency with other "
+			"collision hooks."},
+		{"Asteroid", "object", "Same as \"Object\""},
+		{"Hitpos", "vector", "The world position where the colission was detected"},
+		{"Debris", "object", "The debris object with which the ship collided (only set for debris collisions)"},
+		{"Ship", "ship", "The ship object with which the ship collided (only set for ship collisions)"},
+		{"Weapon", "weapon", "The weapon object with which the ship collided (only set for weapon collisions)"},
+		{"Beam", "weapon", "The beam object with which the ship collided (only set for beam collisions)"} });
+
+const std::shared_ptr<OverridableHook<CollisionConditions>> OnBeamCollision = OverridableHook<CollisionConditions>::Factory("On Beam Collision",
```

Comment:
(on the next line) ship -> beam

(on the hitpos line) "collision"

this needs several rewrites of ship -> beam

for beam collisions, I don't think "ShipB" is applicable; this will not be invoked twice because there are no beam-on-beam collisions; "Ship" is not the same as "Object", and "Beam" is the same as object